private/bufpkg/bufanalysis/print.go

```diff
@@ -56,6 +59,25 @@ func printAsGithubActions(writer io.Writer, fileAnnotations []FileAnnotation) er
 	)
 }
 
+func printAsGitLabCodeQuality(writer io.Writer, fileAnnotations []FileAnnotation) error {
+	report := make([]externalGitLabCodeQualityIssue, 0, len(fileAnnotations))
+	for _, f := range fileAnnotations {
+		if f == nil {
+			continue
+		}
+		fileAnnotationData, err := json.Marshal(newExternalFileAnnotation(f))
+		if err != nil {
+			return err
+		}
+		hash, err := shake256.NewDigestForContent(bytes.NewReader(fileAnnotationData))
+		if err != nil {
+			return err
+		}
```

Comment:
Would move the fingerprint logic into `newExternalGitLabCodeQualityIssue` (you may need to have it error return). Also should this should hash it's contents, not the original FileAnnotation?

Reply:
Moved and made it so that its a hash of the content (minus the hash itself).